src/components/charts/ReferencesTable.tsx

```diff
@@ -0,0 +1,108 @@
+import _ from 'lodash';
+import React, { useContext, useState, useEffect } from "react";
+import { Table, Pagination } from "semantic-ui-react";
+import { AppContext } from "../../context";
+import './Charts.css';
+
+export default function ReferencesTable() {
+  const [column, setColumn] = useState('');
+  const [state, dispatch] = useContext(AppContext);
+  const [activePage, setActivePage] = useState(1);
+  const [boundaryRange, setBoundaryRange] = useState(1);
+  const [siblingRange, setSiblingRange] = useState(1);
+  const [totalPages, setTotalPages] = useState(10);
+
+    const handlePaginationChange = (e : any, event: any) => { 
+      const { activePage } = event;
+      setActivePage(activePage)
+    }
+
+
+  const [direction, setDirection] = useState('ascending');
+  const [data, setData] = useState(state.filtered_records);
+
+  const handleSort = (clickedColumn: string) => () => {
+    if (column !== clickedColumn) {
+      setColumn(clickedColumn);
+      setDirection('ascending');
+      return
+    }
+
+    setDirection(direction === 'ascending' ? 'descending' : 'ascending');
+  }
+
+  useEffect(() => {
+    const newData = _.sortBy(state.filtered_records, [column]);

```

Comment:
maybe we should push all useEffect functions to the top of the component (right below useState/useContext) for better visibility